File: src/main/java/com/chronosx/cx_shop/dtos/OrderDto.java
package com.chronosx.cx_shop.dtos;

import java.time.LocalDate;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.*;
import lombok.experimental.FieldDefaults;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@FieldDefaults(level = AccessLevel.PRIVATE)
public class OrderDto {
    @JsonProperty("user_id")
    @Min(value = 1, message = "User id must be greater than 0")
    Long userId;

    @JsonProperty("full_name")
    String fullName;

    String email;

    @JsonProperty("phone_number")
    @NotBlank(message = "Phone number can't be empty")
    String phoneNumber;

    String address;
    String note;

    @JsonProperty("total_money")
    @Min(value = 0, message = "Total money must be greater than or equal to 0")
    Float totalMoney;

    @JsonProperty("shipping_method")
    String shippingMethod;

    @JsonProperty("shipping_address")
    String shippingAddress;

    @JsonProperty("shipping_date")
    LocalDate shippingDate;

    @JsonProperty("payment_method")
    String paymentMethod;
}

File: cx_shop_fe/src/app/dtos/order/order.dto.ts
export class OrderDto {
    user_id: number;
    full_name: string;
    email: string;
    phone_number: string;    
    address: string;    
    note: string;
    total_money: number;
    payment_method: string;    
    shipping_method: string;
    coupon_code: string;    
    cart_items: { product_id: number; quantity: number }[];

    constructor(data: any) {
        this.user_id = data.user_id;
        this.full_name = data.full_name;
        this.email = data.email;
        this.phone_number = data.phone_number;
        this.address = data.address;
        this.note = data.note;
        this.total_money = data.total_money;
        this.shipping_method = data.shipping_method;
        this.payment_method = data.payment_method;
        this.coupon_code = data.coupon_code;
        this.cart_items = data.cart_items;
    }
}